hack/preview.sh

```diff
@@ -105,14 +105,7 @@ yq -i e ".data |= .\"transparency.url\"=\"https://$rekor_server\"" $ROOT/compone
-[ -n "${JVM_BUILD_SERVICE_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"hacbs-jvm-operator\")) |=.newName=\"${JVM_BUILD_SERVICE_IMAGE_REPO}\"" $ROOT/components/jvm-build-service/kustomization.yaml
-[ -n "${JVM_BUILD_SERVICE_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"hacbs-jvm-operator\")) |=.newTag=\"${JVM_BUILD_SERVICE_IMAGE_TAG}\"" $ROOT/components/jvm-build-service/kustomization.yaml
-[ -n "${JVM_BUILD_SERVICE_CACHE_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"hacbs-jvm-cache\")) |=.newName=\"${JVM_BUILD_SERVICE_CACHE_IMAGE_REPO}\"" $ROOT/components/jvm-build-service/kustomization.yaml
-[ -n "${JVM_BUILD_SERVICE_CACHE_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"hacbs-jvm-cache\")) |=.newTag=\"${JVM_BUILD_SERVICE_CACHE_IMAGE_TAG}\"" $ROOT/components/jvm-build-service/kustomization.yaml
-[ -n "${JVM_BUILD_SERVICE_SIDECAR_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"run-maven-component-build\")) |=.newName=\"${JVM_BUILD_SERVICE_SIDECAR_IMAGE_REPO}\"" $ROOT/components/jvm-build-service/kustomization.yaml
-[ -n "${JVM_BUILD_SERVICE_SIDECAR_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"run-maven-component-build\")) |=.newTag=\"${JVM_BUILD_SERVICE_SIDECAR_IMAGE_TAG}\"" $ROOT/components/jvm-build-service/kustomization.yaml
-[[ -n "${JVM_BUILD_SERVICE_SIDECAR_IMAGE_REPO}" && ${JVM_BUILD_SERVICE_SIDECAR_IMAGE_TAG} ]] && yq -i e "(.spec.template.spec.containers[].env[] | select(.name==\"JVM_BUILD_SERVICE_SIDECAR_IMAGE\")) |=.value=\"${JVM_BUILD_SERVICE_SIDECAR_IMAGE_REPO}:${JVM_BUILD_SERVICE_SIDECAR_IMAGE_TAG}\"" $ROOT/components/jvm-build-service/operator-images.yaml
-[[ -n "${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_REPO}" && ${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_TAG} ]] && yq -i e "(.spec.template.spec.containers[].env[] | select(.name==\"JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE\")) |=.value=\"${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_REPO}:${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_TAG}\"" $ROOT/components/jvm-build-service/operator-images.yaml
+[[ -n "${JVM_BUILD_SERVICE_CACHE_IMAGE}" && ${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE} ]] && patch=$(yq "(.patches[] | select(.target.kind == \"Deployment\")).patch" $ROOT/components/jvm-build-service/kustomization.yaml | yq e "select(.[].path == \"/spec/template/spec/containers/0/env\") | .[].value |= . + [{\"name\" : \"JVM_BUILD_SERVICE_CACHE_IMAGE\", \"value\": \"${JVM_BUILD_SERVICE_CACHE_IMAGE}\"}, {\"name\": \"JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE\", \"value\": \"${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE}\"}] | (.[].value[] | select(.name == \"IMAGE_TAG\")) |= .value = \"\"") yq -i e "(.patches[] | select(.target.kind == \"Deployment\")).patch |= strenv(patch)" $ROOT/components/jvm-build-service/kustomization.yaml
```

Comment:
this might be a bit hardcode one-liner. any suggestions for improvement are welcome 😅

Reply:
ok then we'd have to update the [script for updating jvm-build-service image tag](https://github.com/redhat-appstudio/jvm-build-service/blob/f1ccc0358907c305dc0fe9b5105f17c9a0592755/.tekton/controller-push.yaml#L23), right?

Reply:
updated, [result](https://github.com/psturc/infra-deployments/commit/668d3d2456a862362dce5da65121e7f2a3ae02da#diff-44030e8cb4fc8790e75c962d72e7a46457f9c4ed86e6fd554d206db955dc5685)